File: 13th Dec- Consecutive 1's not allowed/Consecutive1snotallowed.java
//{ Driver Code Starts
// Initial Template for Java

import java.io.*;
import java.util.*;

class Consecutive1snotallowed {
    // Driver code
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine().trim());
        while (t-- > 0) {
            int n = Integer.parseInt(br.readLine().trim());

            long ans = new Solution().countStrings(n);

            System.out.println(ans);
        }
    }
}
// } Driver Code Ends

// User function Template for Java

class Solution {
    long countStrings(int n) {
        // code here
        if (n == 1)
            return 2;
        long one = 1, zero = 1, total = one + zero;

        while (n-- > 1) {
            one = zero;
            zero = total;
            total = (one + zero) % 1000000007;
        }

        return total % 1000000007;
    }
}

File: 13th Dec- Consecutive 1's not allowed/Consecutive1snotallowed.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>

using namespace std;
#define ll long long

// } Driver Code Ends
//User function template for C++



class Solution{
public:
	// #define ll long long
	 int mod=1e9+7;
    ll countStrings(int n) {
        ll f=1;
        ll s=2;
        
        for(int i=2;i<=n;i++){
            ll t=(f+s)%mod;
            f=s;
            s=t;
        }
        return s;
	}
};


int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;
        Solution ob;
        auto ans = ob.countStrings(n);
        cout << ans << "\n";
    }
    return 0;
}
// } Driver Code Ends